### PR TITLE
Support Optimism in Set.js

### PR DESCRIPTION
**Pull request by williamsfu99**
* Supported Chain ID for 10 (Optimism mainnet)
* Support Optimism tokenList from Coingecko
* Support 0x API endpoint for Optimism
* Support gas price fetching for Optimism RPC
* Bump package.json

### CODE REVIEW

**Review by justinkchen — COMMENTED** [review]
lets also use the polygon tokenlist to replace that weird mapping logic:

https://tokens.coingecko.com/polygon-pos/all.json

the rest LGTM i think

File: src/api/utils/gasOracle.ts [hunk 1]
```diff
@@ -75,6 +79,12 @@ export class GasOracleService {
     }
   }
 
+  private async getOptimismGasPrice(): Promise<number> {
+    const price =  await this.provider.getGasPrice();
+
+    return price.toNumber();
+  }
```
Comment (cgewecke) [inline src/api/utils/gasOracle.ts:86]: It turns out there's a url endpoint to get the optimism gas price via their etherscan site here. 

https://api-optimistic.etherscan.io/api?module=proxy&action=eth_gasPrice

If we used that we wouldn't need to pass in the provider / change the instantiation flow here.

**Review by cgewecke — APPROVED** [review]
Nice! Awesome to remove that polygon token list mess too. 

Just left a note about a possible alternative for fetching the optimism gas price but otherwise everything lgtm.

File: src/api/utils/gasOracle.ts (hunk 1 above)
Comment (williamsfu99) [reply src/api/utils/gasOracle.ts:86]: Thanks for sharing! wasn't aware of this. I'll use that instead.

Comment (justinkchen) [reply src/api/utils/gasOracle.ts:85]: one comment is that adding an api endpoint means we have to update our CSP allow list on our UI

**Pull request merged**